jwst/stpipe/tests/test_step.py

```diff
@@ -113,14 +111,17 @@ def throw_error(self):
         "data/single_member_miri_asn.json", package="jwst.stpipe.tests"
     )
 
-    data = single_member_asn
     if on_disk_status is not None:
         with ModelLibrary(single_member_asn, on_disk=on_disk_status) as model_library:
-            data = model_library
+            pars = WhiteLightStep.get_config_from_reference(model_library)
+    else:
+        pars = WhiteLightStep.get_config_from_reference(single_member_asn)
 
-    pars = WhiteLightStep.get_config_from_reference(data)
     assert pars == WHITELIGHTSTEP_CRDS_MIRI_PARS
 
+    if on_disk_status is True:
```

Comment:
@braingram even after modifying this test to (hopefully) ensure everything to do with the `ModelLibrary` happens inside a `with` block, I still need to explicitly clean up the temporary directory to avoid `ResourceWarning`s.  Why is that, and does this represent a small bug in the `exit` method of `ModelLibrary`?  Should it be calling `cleanup()`?

Reply:
`__exit__` "closes" the library for "lending" but not "closes" in the sense that resources are freed. For this test `model_library` is still in scope after the `__exit__` (put another way the `model_library.__enter__` could be called a second time). The `_temp_dir` cleanup occurs during `__del__` which is not called within the body of the test (which leads to the failure on 3.14.2). So I don't think there's a bug in `__exit__` (it needs to keep the temporary directory around for the life of the library).

Testing locally adding `del model_library` instead of `model_library._temp_dir.cleanup` allows these tests to pass (I didn't run the full unit test suite, just the ones in this file).

Reply:
Given the variability in this. If this PR fixes the failures I suggest we go forward with the changes here and any further changes from investigation can be in follow-up PRs.